src/Foreign/Object.purs

```diff
@@ -221,6 +222,15 @@ fromFoldable l = runST do
   ST.foreach (A.fromFoldable l) \(Tuple k v) -> void $ OST.poke k v s
   pure s
 
+-- | Create an `Object a` from an indexed foldable collection, using the
+-- | specified function for converting the index to a `String` key. On key
+-- | collisions, the last value for the duplicate key will be kept.
+fromFoldableWithIndex :: forall f k v. FoldableWithIndex k f => (k -> String) -> f v -> Object v
+fromFoldableWithIndex f l = runST do
+  s <- OST.new
+  forWithIndex_ l \k v -> OST.poke (f k) v s
+  pure s
+
```

Comment:
Thoughts on adding a variant `fromFoldableWithIndex'` that assumes the `k` is already a `String`?

Reply:
I'm now thinking that makes sense as the default API, based on the two use cases[^1] cited in the issue. They both say that `Object.fromFoldableWithIndex` would be useful specifically in the case of `FoldableWithIndex k f ~ Map String`.
```haskell
fromFoldableWithIndex :: forall f a. FoldableWithIndex String f => f a -> Object a
```
I think non-`String` keys would probably be the less common case, and in that case you might as well _also_ throw in a function for combining values on conflict (instead of defaulting to overriding earlier values, which could be undesirable)
```haskell
fromFoldableWithIndexWith :: forall f k v. FoldableWithIndex k f => (k -> String) -> (v -> v -> v) -> f v -> Object v
```
basically support the (probably) common case with `fromFoldableWithIndex` and then give more granular control with `fromFoldableWithIndexWith` (not sure about that naming 😄 )

[^1]: https://discord.com/channels/864614189094928394/865617619464749081/887326007105814548
      https://discord.com/channels/864614189094928394/867149806178664478/1032345440248668242

Reply:
I think we can skip `fromFoldableWithIndexWith` because you could `mapKeys` (or equivalent) the input first.

It would be more efficient to convert while doing this, but you could say the same about accepting mapping functions in `fromFoldable`s in general if we wanted to start going down that route.

Reply:
> It would be more efficient to convert while doing this, but you could say the same about accepting mapping functions in `fromFoldable`s in general if we wanted to start going down that route.

Didn't see your comment til after I posted mine, that makes sense to skip `fromFoldableWithIndexWith`. So would just adding
```haskell
fromFoldableWithIndex :: forall f a. FoldableWithIndex String f => f a -> Object a
```
make sense?